File: DATA STRUCTURES/sump /selectionsort.c
#include<stdio.h>
int* sel(int arr[],int n){
for(int  i = 0; i < n-1; i++)
{
    int small=i;
    for (int  j = i+1; j <n; j++)
    {
       if(arr[j]<arr[small]){
           int temp =arr[j];
           arr[j]=arr[small];
           arr[small]= temp;

       }
    }
    

    
}
    return arr;

}
int main(){
    int arr[]={7,8,9,6,4};
    int *arrs=sel(arr,5);

    for (int i = 0; i < 5; i++)
    {
        printf("%d ",arr[i]);
    }
    
    return 0;

}

File: JAVA/Code/experiments/abdulbari/selectionsort.java

    public class selectionsort {
        public static int[] Ssort(int arr[]) {
    
            for (int i = 0; i < arr.length-1; i++) {
                int small=i;
                for (int j = i+1; j < arr.length; j++) {
                    if(arr[j]<arr[small]){
                        small=j;
                    }
                }
                int temp=arr[i];
                arr[i]=arr[small];
                arr[small]=temp;
            }
            return arr;
        }
    
        public static void main(String[] args) {
            int arr[] = { 8, 57, 7, 34, 26};
            int brr[] = Ssort(arr);
            for (int i = 0; i < brr.length; i++) {
                System.out.print(brr[i]+" ");
            }
        }
    }
    

